doc/vlan_stacking/vlan_stacking_HLD.md

```diff
@@ -0,0 +1,862 @@
+
+```
+admin@sonic:~$ sudo config vlan-translation del -h
+Usage: config vlan-translation del [OPTIONS] <interface_name> <s_vlanid>
+
+  Delete vlan stacking rule, examples:
+  - config vlan-stacking del Ethernet0 200
+
+Options:
+  -?, -h, --help  Show this message and exit.
+```
+
+Display VLAN translation rule configuration
+
+```
+show vlan-translation
+```
+The display format is in the following
+
+```
+admin@root:~# show vlan-translation
+Interface    s_vlanid  c_vlanid
+-----------  --------  --------
+Ethernet0         100       200
+```
+
+### Yang model
+
+New yang model `sonic-vlan-stacking.yang` is defined to describe VLAN stacking configuration.
+
+```json
+module sonic-vlan-stacking {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-vlan-stacking";
+    prefix vlan-stacking;
+
+    import sonic-extension {
+        prefix ext;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+        revision-date 2019-07-01;
+    }
+
+    description "VLAN Stacking YANG Module for SONiC OS";
+
+    revision 2021-12-15 {
+        description "First Revision";
+    }
+
+    container sonic-vlan-stacking {
+        container VLAN_STACKING {
+            description "VLAN_STACKING part of config_db.json";
+
+            list VLAN_STACKING_LIST {
+                key "interface_name s_vlanid";
+
+                ext:key-regex-configdb-to-yang "^([a-zA-Z0-9_-]+)|([a-zA-Z0-9_-]+)|([0-9]+)$";
+
+                ext:key-regex-yang-to-configdb "<interface_name>|<s_vlanid>";
+
+                leaf interface_name {
+                    type union {
+                        type leafref {
+                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:port_name;
+                        }
+                        type leafref {
+                            path /lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:portchannel_name;
+                        }
+                    }
+                }
+
+                leaf s_vlanid {
+                    type leafref {
+                        path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid;
+                    }
+                }
+
+                leaf-list c_vlanids {
+                    type union {
+                        type leafref {
+                            path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid;
+                        }
+                        type string {
+                            length 1..10;
+                            pattern '[0-9]{1,4}..[0-9]{1,4}';
+                        }
+                    }
+                }
+
+                leaf s_vlan_priority {
+                    type uint8 {
+                         range 0..7;
+                    }
```

Comment:
default behaviour?

Reply:
@venkatmahalingam The default behavior is override.